spark/src/main/scala/io/delta/sql/parser/DeltaSqlParser.scala

```diff
@@ -76,6 +77,8 @@ class DeltaSqlParser(val delegate: ParserInterface) extends ParserInterface {
 
   override def parsePlan(sqlText: String): LogicalPlan = parse(sqlText) { parser =>
     builder.visit(parser.singleStatement()) match {
+      case clusterByPlan: ClusterByPlan =>
```

Comment:
Sorry, not obvious to me. How does the Spark returns a `ClusterByPlan`? Can a user type `CREATE TABLE ... CLUSTER BY` using the Spark 3.5 which the Delta depends on?

Reply:
The `ClusterByPlan` is injected by Delta in L416 below. We are using this workaround since Spark 3.5 doesn't support `CLUSTER BY` yet.

The flow looks like this:
1. User types `CREATE TABLE ... CLUSTER BY`.
2. `CLUSTER BY` gets matched with `clusterBySpec` in `DeltaSqlBase.g4` and `visitClusterBy` is called.
3. `visitClusterBy` will create a `ClusterByPlan` which gets detected in `parsePlan` and calls `ClusterByParserUtils.parsePlan`
4. `ClusterByParserUtils.parsePlan` will replace `CLUSTER BY` with `PARTITIONED BY` and call SparkSqlParser(`delegate.parsePlan`) for validation.
5. After successful parsing, clustering column is saved in table properties.